### PR TITLE
Switches to always issue onError with throwable

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This helps prevent subtle bugs. It will take a bit to write tests for these.

### CODE REVIEW

File: zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java [hunk 1]
```diff
@@ -86,11 +86,14 @@ public Factory(HttpClient httpClient) {
...
         try {
-          callback.onSuccess(parseResponse(response, bodyConverter));
-        } catch (IOException e) {
-          callback.onError(e);
+          value = parseResponse(response, bodyConverter);
+        } catch (Throwable t1) {
+          propagateIfFatal(t1);
+          callback.onError(t1);
         }
+        if (value != null) callback.onSuccess(value);
```
Comment (anuraaga) [inline zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java]: Think this needs to go right below line 91. `parseResponse` can return `null` (line 133), which will cause this future to never complete.

**Comment by codefromthecrypt** [comment]
thanks.. good eye. I will also make a test to prove it.

**Comment by anuraaga** [comment]
Took care of it so it can get back to the CI

**Comment by codefromthecrypt** [comment]
thanks. actually I need to pause until monday so if you want to merge
without tests I can backfill monday

**Comment by anuraaga** [comment]
There are tests :)

**Pull request merged**